File: ENGR432/Labs/Lab5/Lab5.py
import random
import time

# Matrix dimensions
M = 1000
N = 1000
K = 1000

# Create two matrices of random integers
matrix_a = [[random.randint(1, 10) for j in range(K)] for i in range(M)]
matrix_b = [[random.randint(1, 10) for j in range(N)] for i in range(K)]

# Create an empty result matrix
result_matrix = [[0 for j in range(N)] for i in range(M)]

# Start the timer
start_time = time.time()

# Multiply the matrices
for i in range(M):
    print('\n')
    for j in range(N):
        for k in range(K):
            result_matrix[i][j] += matrix_a[i][k] * matrix_b[k][j]
            print(str(result_matrix[i][j]),  end=', ')

# End the timer
end_time = time.time()

# Print the elapsed time
print("Elapsed time: %.3f seconds" % (end_time - start_time))
print(result_matrix)

File: ENGR432/Labs/Lab5/Lab5.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

#define ROWS 1000
#define COLS 1000

void generateRandomMatrix(int matrix[ROWS][COLS]) {
    for (int i = 0; i < ROWS; i++) {
        for (int j = 0; j < COLS; j++) {
            matrix[i][j] = rand() % 100;
        }
    }
}

void multiplyMatrices(int matrix1[ROWS][COLS], int matrix2[ROWS][COLS], int result[ROWS][COLS]) {
    for (int i = 0; i < ROWS; i++) {
        for (int j = 0; j < COLS; j++) {
            result[i][j] = 0;
            for (int k = 0; k < COLS; k++) {
                result[i][j] += matrix1[i][k] * matrix2[k][j];
            }
        }
    }
}

int main() {
    int matrix1[ROWS][COLS], matrix2[ROWS][COLS], result[ROWS][COLS];
    srand(time(NULL));

    // Generate random matrices
    generateRandomMatrix(matrix1);
    generateRandomMatrix(matrix2);

    // Multiply matrices
    multiplyMatrices(matrix1, matrix2, result);

    // Print result matrix
    for (int i = 0; i < ROWS; i++) {
        for (int j = 0; j < COLS; j++) {
            printf("%d ", result[i][j]);
        }
        printf("\n");
    }

    return 0;
}